docs/best-practices/crate-node.txt

```diff
@@ -0,0 +1,460 @@
+  it impossible to use some or all of the tables in the cluster.
+
+The sequence of operations for using this tool are as follows:
+
+* Make sure you have really lost access to at least half of the master-eligible
+  nodes in the cluster, and they cannot be repaired or recovered by moving their
+  data paths to healthy hardware.
+
+* Stop all remaining nodes.
+
+* Choose one of the remaining master-eligible nodes to become the new elected
+  master as described above.
+
+* On this node, run the ``crate-node unsafe-bootstrap`` command as shown
+  below. Verify that the tool reported Master node was successfully bootstrapped.
+
+* Start this node and verify that it is elected as the master node.
+
+* Run the ``crate-node detach-cluster`` tool, described below, on every other
+  node in the cluster.
+
+* Start all other nodes and verify that each one joins the cluster.
+
+* Investigate the data in the cluster to discover if any was lost during this
+  process.
+
+* When you run the tool it will make sure that the node that is being used
+  to bootstrap the cluster is not running. It is important that all other
+  master-eligible nodes are also stopped while this tool is running, but
+  the tool does not check this.
+
+The message `Master node was successfully bootstrapped` does not mean that
+there has been no data loss, it just means that tool was able to complete its job.
+
+Detach a node from a cluster
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When a node joins a cluster initially, the unique cluster id will be stored in
+its nodes metadata. This prevents the node from joining a cluster with a different
+cluster id. There are situations where it necessary to reset the cluster id from
+a node. When a cluster is not recoverable, it might be worth moving nodes
+to a new cluster. This is in particular useful when the voting configuration
+of a node is reset, and a new cluster is bootstrapped.
+
+It is unsafe for nodes to move between clusters, because different clusters have
+completely different cluster metadata. There is no way to safely merge the
+metadata from two clusters together.
+
+To protect against inadvertently joining the wrong cluster, each cluster creates
+a unique identifier, known as the cluster UUID, when it first starts up. Every
+node records the UUID of its cluster and refuses to join a cluster with a
+different UUID.
+
+However, if a node’s cluster has permanently failed then it may be desirable
+to try and move it into a new cluster. The ``crate-node detach-cluster`` command
+lets you detach a node from its cluster by resetting its cluster UUID.
+It can then join another cluster with a different UUID.
+
+For example, after unsafe cluster bootstrapping you will need to detach all
+the other surviving nodes from their old cluster so they can join the new,
+unsafely-bootstrapped cluster.
+
+Unsafe cluster bootstrapping is only possible if there is at least one surviving
```

Comment:
Wouldn't go into unsafe bootstrapping details again here. Could refer to the previous section.